control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -114,6 +115,36 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
 	return nil
 }
 
+func getConditionByType(conditions []metav1.Condition, conditionType string) *metav1.Condition {
```

Comment:
couldn't we reuse built in funcs https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L25-L95?

Reply:
Also if we need adhoc funcs to manipulate conditions we probably want to export them so we can reuse them across CRs